File: programming/Programmers/c++/문자열을 정수로 바꾸기.cpp
/*
프로그래머스 문자열을 정수로 바꾸기
-일 경우, +일 경우, 부호가 없을 경우 3가지로 나누어서 stoi를 사용
*/
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
	int answer = 0;
	if (s[0] == '-') {
		answer = -1 * stoi(s.substr(1, s.length() - 1));
	}
	else if (s[0] == '+') {
		answer = stoi(s.substr(1, s.length() - 1));
	}
	else {
		answer = stoi(s.substr(0, s.length()));
	}
	return answer;
}

File: programming/Programmers/javascript/문자열을 정수로 바꾸기.js
/*
프로그래머스 문자열을 정수로 바꾸기
-일 경우 숫자를 제외한 모든 문자를 삭제 후 정수 변환 및 음수 변환
아닐 경우 숫자를 제외한 모든 문자를 삭제 후 정수 변환
*/
function solution(s) {
    var answer = 0;
    for(var i=0; i<s.length; i++){
        if(s.charAt(0)=='-'){
            answer=s.replace(/[^0-9]/g,'');
            answer=parseInt(answer);
            answer=answer*-1;
        }else{
            answer=s.replace(/[^0-9]/g,'');
            answer=parseInt(answer);
        }
    }
    return answer;
}
